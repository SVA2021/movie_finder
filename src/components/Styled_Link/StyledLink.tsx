import {Link, Typography} from '@mui/material';
import {FC} from 'react';

interface StyledLinkProps {
  href: string
  title: string
}

export const StyledLink: FC<StyledLinkProps> = ({href, title}) => {
  return (
    <Link
      data-testid={'styled-link'}
      component={'a'}
      target={'_blank'}
      rel={'noopener noreferrer'}
      href={href}
      underline={'hover'}
    >
      <Typography
        m={1}
        p={2}
        variant="body1"
        component="p"
        textAlign={'center'}
      >
        {title}
      </Typography>
    </Link>
  );
};