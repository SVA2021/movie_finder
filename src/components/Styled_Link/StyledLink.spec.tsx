import {render, screen} from "@testing-library/react";
import {StyledLink} from "./StyledLink";

describe('render StyledLink', () => {

  const setUp = () => render(<StyledLink href={'fakehref'} title={"fakeTitle"} />);

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('styled-link')).toMatchSnapshot();
  });

  it('check title', () => {
    setUp();
    expect(screen.getByText(/fakeTitle/i)).toBeInTheDocument();
  });

  it('check link href', () => {
    setUp();
    expect(screen.getByRole('link')).toHaveAttribute('href', 'fakehref')
  });

})