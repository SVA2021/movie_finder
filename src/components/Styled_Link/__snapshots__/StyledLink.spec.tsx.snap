// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render StyledLink snapshot test 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-17dcqu8-MuiTypography-root-MuiLink-root"
  data-testid="styled-link"
  href="fakehref"
  rel="noopener noreferrer"
  target="_blank"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-137b474-MuiTypography-root"
  >
    fakeTitle
  </p>
</a>
`;
